src/tests/nested-containers.test.ts

```diff
@@ -0,0 +1,650 @@
+
+        // Test action for pair< tuple< uint16_t, uint16_t >>
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('alice').setpt('alice', {'key':10, 'value':[100,101]}) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+
+        // Test action for tuple< uint16_t, set< uint16_t >, set< uint16_t >>
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('alice').settst('alice', [10,[21,31], [41,51,61]]) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+
+        // Test action for tuple< uint16_t, vector< uint16_t >, vector< uint16_t >
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('alice').settv('alice', [16,[26,36], [46,506,606]]) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+
+        // Test action for tuple< optional< uint16_t >, optional< uint16_t >, optional< uint16_t > , optional< uint16_t >, optional< uint16_t >>
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('alice').setto('alice', [100, null, 200, null, 300]) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('bob').setto('bob', [null, null, 10, null, 20]) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+
+        // Test action for tuple< map< uint16_t, uint16_t >, map< uint16_t, uint16_t >>
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('alice').settm('alice', [126, [{'key':10,'value':100},{'key':11,'value':101}], [{'key':80,'value':800},{'key':81,'value':9009}] ]) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+
+        // Test action for tuple< uint16_t, pair< uint16_t, uint16_t >, pair< uint16_t, uint16_t >>
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('alice').settp('alice', [127, {'key':18, 'value':28}, {'key':19, 'value':29}]) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+
+        // Test action for tuple< tuple< uint16_t, uint16_t >, tuple< uint16_t, uint16_t >, tuple< uint16_t, uint16_t >>
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('alice').settt('alice', [[1,2],[30,40], [50,60]]) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+
+        // Test action for vector<optional<mystruct>>
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('alice').setvos('alice', [{'_count': 18, '_strID': 'dumstr'}, null, {'_count': 19, '_strID': 'dumstr'}]) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+
+        // Test action for pair<uint16_t, vector<optional<uint16_t>>>
+        await api.transact({
+            actions: [ api.with('nestcontnmi').as('alice').setpvo('alice',{'first': 183, 'second':[100, null, 200]}) ]
+        }, { blocksBehind: 3, expireSeconds: 30 });
+
+        const transaction = await rpc.get_table_rows({
+            code: 'nestcontnmi',
```

Comment:
Great, 'get table' is important, those setxx actions will pack/Write data into nodeos block-chain, get table and get kv table will unpack/Read data from nodeos, I just want to make sure they are used explicitly, no correction action